src/pages/MarketPage/components/SalesProducts/SalesProducts.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import "./SalesProducts.css";
+import { getProducts } from "../../../../api/api";
+import Product from "../Product/Product";
+import { Link } from "react-router-dom";
+import { CiSearch } from "react-icons/ci";
+import { IoMdArrowDropdown } from "react-icons/io";
+
+function SalesProducts() {
+  const [products, setProducts] = useState([]);
+  const [isSelectVisibleList, setIsSelectVisibleList] = useState(false);
+  const [orderBy, setOrderBy] = useState("recent");
+  const [pageSize, setPageSize] = useState(10);
+
+  const fetchSalesProducts = async (pageSize, orderBy) => {
+    const { list: productList } = await getProducts(pageSize, orderBy);
+    setProducts(productList);
+  };
+  const getHtmlSize = () => {
+    const width = window.innerWidth;
+    if (width < 768) {
+      return 4;
+    } else if (width < 1200) {
+      return 6;
+    } else {
+      return 10;
+    }
+  };
+  useEffect(() => {
+    fetchSalesProducts(10, orderBy);
+    const handleResizeScreen = () => setPageSize(getHtmlSize);
```

Comment:
`getHtmlSize`를 직접 호출하는게 좀 더 직관적이지 않을까요? 

```suggestion
    const handleResizeScreen = () => setPageSize(getHtmlSize());
```

Reply:
아 이 부분 오타였습니다 ㅎㅎ
그리고 그 원식님 pr 답볍해주신걸보다가
resize이벤트 리스너의 콜백함수를 useEffect밖에다가 정의하는게 좋다고 하셨는데 저도 useEffect안에서 정의해서
왜 그런지 궁금합니다. (useEffect안에다 이벤트리스너 콜백함수를 정의한이유는 딱히 없긴합니다 그냥 검색해보니 거의 다 
useEffect안에서 정의 해서 따라서 안에다 정의했습니다.)

Reply:
최적화와 관련이 있습니다.  
useEffect 의 콜백함수가 실행이 될때마다 함수가 재정의되는데 엄청 대단히 성능에 영향을 끼치는건 아니지만 이론적으로는 
지속적으로 재정이되면 좋지 않은건 사실입니다. 그래서 이런 부분을 해결하고자 리액트는 최적화 할 수 있는 훅들을 만들어 놓았습니다. 

그런데 함수가 useEffect 내부에 있으면 최적화용 훅을 사용 할 수 있는 방법이 없습니다. 

`useCallback`,`useMemo` 훅에 대해서 한번 알아보시면 좋을 것 같습니다 ㅎㅎ